File: 5-Problem-A-Day/6. ZigZag Conversion.java
class Solution {
    public String convert(String s, int numRows) {
        List[] rows = new List[numRows];
        int count = 0;
        for (int i = 0; i < numRows; i++)
            rows[i] = new ArrayList<>();        
        int flag = 1;
        for (char c : s.toCharArray()) {            
            rows[count].add(c);
            count = (count + flag) % numRows;
            if (count == 0 || count == numRows - 1)
                flag = -flag;
        }   
        String ans = "";
        for (int i = 0; i < numRows; i++) {
            for (int j = 0; j < rows[i].size(); j++) {
                ans += rows[i].get(j);
            }
        }
        return ans;
    }
}

File: 2023/6. Zigzag Conversion.cpp
class Solution {
public:
    string convert(string s, int numRows) {
         vector<vector<char>> rows(numRows);
         int count = 0, flag = 1;
         for (char c : s) {
             rows[count].push_back(c);
             count = (count + flag) % numRows;
             if (count == 0 || count == numRows - 1) {
                 flag = -flag;
             }
         }
         string res;
         for (vector<char>& row : rows) {
             for (char c : row) {
                 res += c;
             }
         }
         return res;
    }
};